scripts/template_matcher.py

```diff
@@ -16,39 +17,53 @@ def __init__ (self, images, min_match_count=10, good_thresh=0.7):
         else:
             self.sift = cv2.SIFT() #initialize SIFT to be used for image matching
 
-        # for potential tweaking
+        self.bf = cv2.BFMatcher()
         self.min_match_count = min_match_count
-        self.good_thresh = good_thresh #use for keypoint threshold
+        self.good_thresh = good_thresh #use for keypoint threshold        
 
-        #TODO: precompute keypoints for template images
+        # precompute keypoints for template images
```

Comment:
I noticed in another student's code that they made their own `get_template_keypoints` function.  I really liked how it was modularized, while keeping the __init__ function at the same abstraction level.  Check it out and tell me if you think it's nicer to put it in a function!

Credits to Sean.
https://github.com/SeanCCarter/sign_follower/blob/master/scripts/template_matcher.py#L36